jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -369,11 +369,21 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
                                            uncertainty=collapsed_as_mean.uncertainty,
                                            wcs=collapsed_as_mean.wcs,
                                            meta=collapsed_as_mean.meta)
+        elif selected_func == 'sum':
+            collapsed_nddata = getattr(nddata_reshaped, selected_func)(
+                axis=spatial_axes, **kwargs
+            )  # returns an NDDataArray
+            # Remove per steradian denominator
+            if astropy.units.sr in collapsed_nddata.unit.bases:
+                new_unit = collapsed_nddata.unit * astropy.units.sr
+                collapsed_nddata.unit = new_unit
+                if hasattr(collapsed_nddata.uncertainty, 'unit'):
+                    new_uncert_unit = collapsed_nddata.uncertainty.unit * astropy.units.sr
+                    collapsed_nddata.uncertainty.unit = new_uncert_unit
```

Comment:
This conversion needs a multiplication with an area term that has units [sr/pixel]. The pixels have units like [Jy / sr]. If we want to sum over an area and remove the area unit, we need to multiply `(Jy / sr) * (sr / pixel)`. The second term is stored in the `pixar_sr` keyword, and you can get it with:

```
self.spectral_cube.meta.get('PIXAR_SR', 1.0)
```

Reply:
See for example: https://github.com/spacetelescope/jdaviz/pull/2860/files#diff-5835bb5bebcbbcd6d476dd40734390a1c56bb3f1e51403b8dfb78ad8e735dbe6R487

Reply:
you need the area of the aperture itself.  You can just refer to `self.aperture_area_along_spectral` directly to get the area (in pixels) and multiply by the PIXAR_SR to get it in `sr`.

Reply:
`self.aperture_area_along_spectral` is an array, so is the area one of the elements of the array or the elements multiplied together?

Reply:
its an area for each slice (since the aperture could be a cone)

Reply:
I see that the `wavelength_dependent` parameter is not used in `_extract_from_aperture`, does that mean `main` no longer has that functionalioty?

Reply:
It can probably be removed from the method signature, yes.  Wavelength dependence is now being handled by passing `reference_spectral_value` to `get_mask` in the `aperture_weight_mask` and `bg_weight_mask` properties, respectively (although please feel free to test to make sure its working as expected).

Reply:
Hopefully I'm on the right track now. I added a check of the collapsed flux to the test so you can see if the values look correct. The flux from the original collapsed nddata sum array have been multiplied by the `self.aperture_area_along_spectral` values (in the test case they are 20).